flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -81,14 +92,36 @@ NettyPartitionRequestClient createPartitionRequestClient(ConnectionID connection
 				Object old = clients.putIfAbsent(connectionId, connectingChannel);
 
 				if (old == null) {
-					nettyClient.connect(connectionId.getAddress()).addListener(connectingChannel);
-
-					client = connectingChannel.waitForChannel();
-
-					clients.replace(connectionId, connectingChannel, client);
+					int count = 0;
+					ChannelException exception = null;
+					do {
+						try {
+							exception = null;
+							connectingChannel.error = null;
+							LOG.info("Connecting to {} at {} attempt", connectionId.getAddress(), count);
+							nettyClient.connect(connectionId.getAddress()).addListener(connectingChannel);
+
+							client = connectingChannel.waitForChannel();
+
+							clients.replace(connectionId, connectingChannel, client);
+						} catch (ChannelException e) {
+							LOG.error("Failed {} times to connect to {}", count, connectionId.getAddress(), e);
+							exception = e;
+						}
+						count++;
+					} while (count <= retryNumber);
+
+					if (exception != null) {
+						throw exception;
+					}
```

Comment:
Can you extract this into a separate method? This method is already a bit too big.

Reply:
Done